crates/lockfile/src/resolution.rs

```diff
@@ -45,10 +46,10 @@ pub enum LockfileResolution {
 
 impl LockfileResolution {
     /// Get the integrity field if available.
-    pub fn integrity(&self) -> Option<&'_ str> {
+    pub fn integrity(&self) -> Option<&'_ Integrity> {
         match self {
-            LockfileResolution::Tarball(resolution) => resolution.integrity.as_deref(),
-            LockfileResolution::Registry(resolution) => resolution.integrity.as_str().pipe(Some),
+            LockfileResolution::Tarball(resolution) => resolution.integrity.as_ref(),
+            LockfileResolution::Registry(resolution) => resolution.integrity.pipe_ref(Some),
```

Comment:
Is this the same as this?
```suggestion
            LockfileResolution::Tarball(resolution) => &resolution.integrity,
            LockfileResolution::Registry(resolution) => Some(&resolution.integrity),
```

Reply:
@zkochan You're right on the second line. The first line is not the same. [`Option::as_ref`] is a method that converts `&Option<T>` to `Option<&T>`.